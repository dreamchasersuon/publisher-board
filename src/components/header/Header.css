.Header {
  display: flex;
  flex-direction: column;
  padding: 50px 50px 10px 50px;
  border-bottom: 1px solid #e5e5e5;
}

h1 {
  margin: 0;
  font-weight: lighter;
  font-family: 'Montserrat', sans-serif;
}

.Header-title {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.Header-title span {
  color: #666;
}
