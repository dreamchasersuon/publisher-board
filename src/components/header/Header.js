import React from 'react';
import PropTypes from 'prop-types';
import Button from '../buttons/button-open-modal/Button';
import './Header.css';

export default function Header({ handleOnOpenModal, title, caption }) {
  return (
    <header className="Header">
      <div className="Header-title">
        <h1>{title}</h1>
        <span>{caption}</span>
      </div>
      <Button
        onPress={handleOnOpenModal}
        title="Add"
        dataName="addUser"
        className="Header-button"
      />
    </header>
  );
}

Header.propTypes = {
  handleOnOpenModal: PropTypes.func,
  title: PropTypes.string,
  caption: PropTypes.string,
};
