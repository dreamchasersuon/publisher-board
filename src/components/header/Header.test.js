import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

test('renders header with title and caption', () => {
  const tree = renderer
    .create(<Header title="Players" caption="manage users of your game" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
