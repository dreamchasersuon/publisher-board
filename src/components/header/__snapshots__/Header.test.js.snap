// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header with title and caption 1`] = `
<header
  className="Header"
>
  <div
    className="Header-title"
  >
    <h1>
      Players
    </h1>
    <span>
      manage users of your game
    </span>
  </div>
  <button
    className="Header-button"
    data-name="addUser"
  >
    Add
  </button>
</header>
`;
